integration/nodejs_stack_integration_test.go

```diff
@@ -0,0 +1,352 @@
+					settings.Buildpacks.NPMInstall.Online,
+					settings.Buildpacks.BuildPlan.Online,
+				).
+				WithBuilder(builder).
+				WithNetwork("host").
+				WithEnv(map[string]string{"BP_NODE_RUN_EXTENSION": runNodejs18ImageID}).
+				WithPullPolicy("always").
+				Execute(name, source)
+			Expect(err).NotTo(HaveOccurred())
+
+			container, err = docker.Container.Run.
+				WithPublish("8080").
+				WithCommand("node server.js").
+				Execute(image.ID)
+			Expect(err).NotTo(HaveOccurred())
+
+			Eventually(container).Should(Serve("Hello World!"))
+		})
+	})
+
+	context("When using Node.js 20 as run image", func() {
+		var (
+			err                error
+			builder            string
+			buildImageID       string
+			runImageID         string
+			runNodejs20ImageID string
+			nodeMajorVersion   int
+		)
+
+		it.After(func() {
+			Expect(docker.Image.Remove.Execute(builder)).To(Succeed())
+			Expect(docker.Image.Remove.Execute(buildImageID)).To(Succeed())
+			Expect(docker.Image.Remove.Execute(runImageID)).To(Succeed())
+			Expect(docker.Image.Remove.Execute(runNodejs20ImageID)).To(Succeed())
+			Expect(os.RemoveAll(source)).To(Succeed())
+		})
+
+		it.Before(func() {
+			nodeMajorVersion = 20
+
+			buildImageID, runImageID, builderConfigFilepath, builder, err = generateBuilder(root)
+			Expect(err).NotTo(HaveOccurred())
+
+			source, err = occam.Source(filepath.Join("testdata", "nodejs_simple_app"))
+			Expect(err).NotTo(HaveOccurred())
+
+			//Creating and pushing the run image to local registry
+			runNodejs20Archive := filepath.Join(root, fmt.Sprintf("../build-nodejs-%d", nodeMajorVersion), "run.oci")
+			runNodejs20ImageID = fmt.Sprintf("run-nodejs-%d-%s", nodeMajorVersion, uuid.NewString())
+			Expect(err).NotTo(HaveOccurred())
+
+			Expect(archiveToDaemon(runNodejs20Archive, runNodejs20ImageID)).To(Succeed())
+		})
+
+		it("sucessfully builds an app", func() {
+
+			image, _, err = pack.Build.
+				WithExtensions(
+					settings.Extensions.UbiNodejsExtension.Online,
+				).
+				WithBuildpacks(
+					settings.Buildpacks.NodeEngine.Online,
+					settings.Buildpacks.NPMInstall.Online,
+					settings.Buildpacks.BuildPlan.Online,
+				).
+				WithBuilder(builder).
+				WithNetwork("host").
+				WithEnv(map[string]string{"BP_NODE_RUN_EXTENSION": runNodejs20ImageID}).
+				WithPullPolicy("always").
+				Execute(name, source)
+			Expect(err).NotTo(HaveOccurred())
+
+			container, err = docker.Container.Run.
+				WithPublish("8080").
+				WithCommand("node server.js").
+				Execute(image.ID)
+			Expect(err).NotTo(HaveOccurred())
+
+			Eventually(container).Should(Serve("Hello World!"))
+		})
+	})
+
+}
+
+func archiveToDaemon(path, id string) error {
+	skopeo := pexec.NewExecutable("skopeo")
+
+	return skopeo.Execute(pexec.Execution{
+		Args: []string{
+			"copy",
+			fmt.Sprintf("oci-archive:%s", path),
+			fmt.Sprintf("docker-daemon:%s:latest", id),
+		},
+	})
+}
+
+func createBuilder(config string, name string) (string, error) {
```

Comment:
I'm not sure having this in a separate function helps. At first read I wondered what the diff was between createBuilder and generateBuilder. I think I'd just pull this into the generateBuilder function.

Reply:
I agree